src/loader/BlendLoader.py

```diff
@@ -73,28 +106,88 @@ def run(self):
+                    
+                    # setup proeprties. For Mesh based Objects use LoaderInterface._set_properties
+                    # that expects a mesh object and sets the polygon of the mesh to smooth/non smooth
+                    # Non mesh based objects dont have polygons.
+                    if hasattr(added_resource, 'type') and added_resource.type == 'MESH':
+                        self._set_properties([added_resource])
+                    else:
+                        # Non Mesh based object like Light, Camera wrapepd in objects
+                        # or non wrappable objects like materials, textures
+                        self._set_datablock_properties(added_resource)
             else:
                 raise Exception("Unsupported datablock/folder name: " + load_from +
                                 "\nSupported names: " + str(self.known_datablock_names.keys()) +
                                 "\nIf your ID exists, but not supported, please append a new pair of "
                                 "{type ID(folder name): parameter name} to the 'known_datablock_names' dict. Use this "
                                 "for finding your parameter name: " + str(dir(data_from)))
-        newly_loaded_objects = list(set(bpy.context.scene.objects) - previously_loaded_objects)
-        self._set_properties(newly_loaded_objects)
+
+    def _find_datablock_name_match_in_blendfile(self, blend_file_data, data_block_name):
+        """
+        Finds the corresponding datablock name in loaded .blend file.
+        .blend file uses slightly different string name for Datablocks, includes
+        underscores and extra s/'es'. For example /Mesh datablock name is matched to 
+        meshes attribute of .blend file. GreasePencil is matched to grease_pencils and so on.
+
+        :param blend_file_data: contents of loaded .blend file
+        :param data_block_name: Datablock name of a .blend file
+        """
+        blend_file_datablock_names = dir(blend_file_data)
+        index = -1
+        for i, attr in enumerate(blend_file_datablock_names):
+            # remove underscores like grease_pencils -> greasepencils and match with GreasePencil
+            # datablock
+            attr = attr.replace('_', '')
+            if data_block_name.lower() in attr:
+                index = i
+                break
+        if index == -1:
+
+            # The Datablock is valid but the .blend file does not contain the datablock. Likely
+            # version not supported.
+            raise Exception("Could not match Datablock {} in the .blend file. please Verify that \
+                            the .blend file version supports {} ID. Current \
+                             Blender API Version {}".format(data_block_name, data_block_name, bpy.app.version_string))
+
+        return blend_file_datablock_names[index]
+
+    def _set_datablock_properties(self, resource):
```

Comment:
Do you think that maybe modifying the [_set_properties](https://github.com/DLR-RM/BlenderProc/blob/master/src/loader/LoaderInterface.py#L34) method of the parent interface is a better solution than creating a new here? I see this case being general enough for pulling it into the parent class, additionally this mod should be easy enough I think, we just make sure locally that we pass right types there.

Reply:
Yes it would be better to move it there but I avoided not to potentially break other code, as its used by all other loaders.

Reply:
Can you try to make it so its general enough for this example  but work the same for other examples that use BlendLoader?

Reply:
done

Reply:
> done

Then we should use the parent class method. Did you make according changes?

Reply:
yes already updated it in the parent class and removed this method from BlenderLoader